aci/resource_aci_fvesg.go

```diff
@@ -0,0 +1,696 @@
+			err = aciClient.DeleteRelationfvRsIntraEpgFromEndpointSecurityGroup(fvESg.DistinguishedName, relDn)
+			if err != nil {
+				return err
+			}
+
+		}
+
+		for _, relDn := range relToCreate {
+			err = aciClient.CreateRelationfvRsIntraEpgFromEndpointSecurityGroup(fvESg.DistinguishedName, relDn)
+			if err != nil {
+				return err
+			}
+			d.Partial(true)
+			d.SetPartial("relation_fv_rs_intra_epg")
+			d.Partial(false)
+
+		}
+
+	}
+
+	d.SetId(fvESg.DistinguishedName)
+	log.Printf("[DEBUG] %s: Update finished successfully", d.Id())
+
+	return resourceAciEndpointSecurityGroupRead(d, m)
+
+}
+
+func resourceAciEndpointSecurityGroupRead(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning Read", d.Id())
+
+	aciClient := m.(*client.Client)
+
+	dn := d.Id()
+	fvESg, err := getRemoteEndpointSecurityGroup(aciClient, dn)
+
+	if err != nil {
+		d.SetId("")
+		return nil
+	}
+	setEndpointSecurityGroupAttributes(fvESg, d)
+
+	fvRsSecInheritedData, err := aciClient.ReadRelationfvRsSecInheritedFromEndpointSecurityGroup(dn)
+	if err != nil {
+		log.Printf("[DEBUG] Error while reading relation fvRsSecInherited %v", err)
+
+	} else {
+		d.Set("relation_fv_rs_sec_inherited", fvRsSecInheritedData)
+	}
+
+	fvRsProvData, err := aciClient.ReadRelationfvRsProvFromEndpointSecurityGroup(dn)
+	if err != nil {
+		log.Printf("[DEBUG] Error while reading relation fvRsProv %v", err)
+
+	} else {
+		d.Set("relation_fv_rs_prov", fvRsProvData)
+	}
+
+	fvRsConsIfData, err := aciClient.ReadRelationfvRsConsIfFromEndpointSecurityGroup(dn)
+	if err != nil {
+		log.Printf("[DEBUG] Error while reading relation fvRsConsIf %v", err)
+
+	} else {
+		d.Set("relation_fv_rs_cons_if", fvRsConsIfData)
+	}
+
+	fvRsCustQosPolData, err := aciClient.ReadRelationfvRsCustQosPolFromEndpointSecurityGroup(dn)
+	if err != nil {
+		log.Printf("[DEBUG] Error while reading relation fvRsCustQosPol %v", err)
+
+	} else {
+		d.Set("relation_fv_rs_cust_qos_pol", fvRsCustQosPolData)
+	}
+
+	fvRsConsData, err := aciClient.ReadRelationfvRsConsFromEndpointSecurityGroup(dn)
+	if err != nil {
+		log.Printf("[DEBUG] Error while reading relation fvRsCons %v", err)
+
+	} else {
+		d.Set("relation_fv_rs_cons", fvRsConsData)
+	}
+
+	fvRsScopeData, err := aciClient.ReadRelationfvRsScopeFromEndpointSecurityGroup(dn)
+	if err != nil {
+		log.Printf("[DEBUG] Error while reading relation fvRsScope %v", err)
+
+	} else {
+		d.Set("relation_fv_rs_scope", fvRsScopeData)
+	}
+
+	fvRsProtByData, err := aciClient.ReadRelationfvRsProtByFromEndpointSecurityGroup(dn)
+	if err != nil {
+		log.Printf("[DEBUG] Error while reading relation fvRsProtBy %v", err)
+
+	} else {
+		d.Set("relation_fv_rs_prot_by", fvRsProtByData)
+	}
+
+	fvRsIntraEpgData, err := aciClient.ReadRelationfvRsIntraEpgFromEndpointSecurityGroup(dn)
+	if err != nil {
+		log.Printf("[DEBUG] Error while reading relation fvRsIntraEpg %v", err)
+
+	} else {
+		d.Set("relation_fv_rs_intra_epg", fvRsIntraEpgData)
+	}
```

Comment:
There should be some comparison logic for all the relation attribute in READ. The reason for that is some relations use named  attribute in place for tDn, so terraform file will have tDn and you set internally the name and will show change all the time.